pdm/project/config.py

```diff
@@ -195,15 +195,20 @@ def _save_config(self) -> None:
             tomlkit.dump(toml_data, fp)  # type: ignore
 
     def __getitem__(self, key: str) -> Any:
-        env_var = self._config_map[key].env_var
+        try:
+            config = self._config_map[key]
+        except KeyError:
+            # ignoring removed plugin keys
+            return
+        env_var = config.env_var
```

Comment:
I think it would be better to keep the KeyError for `__getitem__` instead of returning a `None`.

Reply:
Raising the KeyError here will fall back to the previous behaviour.  With this traceback
```bash
Traceback (most recent call last):
  File "D:\projects\pdm\pdm\project\config.py", line 199, in __getitem__
    config = self._config_map[key]
KeyError: 'venv.in_project'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "D:\projects\pdm\.venv\Scripts\pdm.exe\__main__.py", line 7, in <module>
  File "D:\projects\pdm\pdm\core.py", line 219, in main
    return Core().main(args)
  File "D:\projects\pdm\pdm\core.py", line 172, in main
    if options.project.config["check_update"]:
  File "D:\projects\pdm\pdm\project\core.py", line 130, in config
    result = dict(self.global_config)
  File "D:\projects\pdm\pdm\project\config.py", line 202, in __getitem__
    raise KeyError
```
Return ignore will still tag invalid key as before.

Reply:
That is because `__iter__()` returns all presented config items, you need to exclude those that can't be recognized.

Reply:
Ok let me check that.

Reply:
So something like this 
```python
def __getitem__(self, key: str) -> Any:
    # check if keys are valid
    if key in (self._data.keys() - self._config_map.keys()):
        return
    config = self._config_map[key]
    env_var = config.env_var
```
will ignore those keys which are presented in the config file but is not a part of valid `_config_map` 
So, `__getitem__` will now still raise a key error if it is not a part of config file.

Reply:
The `KeyError` is desired, don't ignore it.

You need to change `__iter__()` method to exclude keys not in `_config_map`.

Reply:
Done moved the ignore logic to `__iter__()` method